### ISSUE TITLE
Add timeout for tests?

### ISSUE THREAD

**Issue by giltho** [facebookexperimental/reason-native#236]
Adding optional timeouts for tests would be nice (to detect infinite loops for example)

**Comment by giltho** [comment]
Apparently, designing a `timeout` function is quite difficult on non-unix machines (https://github.com/ocaml/ocaml/issues/5908) so I am not sure if this is possible or not.
This is a sad limitation of OCaml right here :/
If anyone has any idea of how one could do that in the case of Rely it would be great, otherwise this can at least serve as documentation for why it isn't implemented in Rely

**Comment by kyldvs** [comment]
Yes this would be a cool feature, but looks challenging based on the limitations you point out.

I wonder if we can work around this by moving tests that have a timeout specified to a separate process which we can kill if it exceeds their timeout. That might be possible even cross platform

**Comment by giltho** [comment]
That could work indeed !
Building up on what you say, maybe this could be done by compiling a tiny bit differently when targeting JS and when targeting native : 
 - Use the node API when targeting native
 - Use [luv](https://github.com/aantron/luv) when targeting native. I think it doesn't have the limitations of Unix.
